x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ChunkedTextEmbeddingResults.java

```diff
@@ -51,6 +51,13 @@ public Map<String, Object> asMap() {
             return map;
         }
 
+        public Map<String, Object> asMapWithListsInsteadOfArrays() {
+            var map = new HashMap<String, Object>();
+            map.put(TEXT, matchedText);
+            map.put(INFERENCE, Arrays.stream(embedding).boxed().collect(Collectors.toList()));
```

Comment:
New function here only for testing to get the double array as a list. `Matcher.is` seem to work when comparing the lists but not arrays 🤷‍♂️